packages/theme-language-server-common/src/linkedEditingRanges/providers/HtmlTagNameLinkedRangesProvider.ts

```diff
@@ -0,0 +1,27 @@
+import { LiquidHtmlNode } from '@shopify/liquid-html-parser';
+import { LinkedEditingRangeParams } from 'vscode-languageserver';
+import { DocumentManager } from '../../documents';
+import { BaseLinkedEditingRangesProvider } from '../BaseLinkedEditingRangesProvider';
+import { getHtmlElementNameRanges } from '../../utils/htmlTagNames';
+
+export class HtmlTagNameLinkedRangesProvider implements BaseLinkedEditingRangesProvider {
+  constructor(public documentManager: DocumentManager) {}
+
+  async linkedEditingRanges(
+    node: LiquidHtmlNode | null,
+    ancestors: LiquidHtmlNode[] | null,
+    params: LinkedEditingRangeParams,
+  ) {
+    if (!node || !ancestors) return null;
+
+    const textDocument = this.documentManager.get(params.textDocument.uri)?.textDocument;
+    if (!textDocument) return null;
+
+    const ranges = getHtmlElementNameRanges(node, ancestors, params, textDocument);
+    if (!ranges) return null;
+    return {
+      ranges,
+      wordPattern: '[a-zA-Z0-9-_]+',
```

Comment:
The `_` char is acting funky when i tophat it.

Once you add a `_` to the tag name, it just doesn't duplicate the rest.

E.g.

```
<!-- works -->
<foo></foo>
<foo-bar></foo-bar>

<!-- halts after typing _ -->
<foo_bar></foo_>
```

`getHtmlElementNameRanges` doesn't return `ranges` when there's an underscore in it.

Reply:
Removed `_` from the word pattern :+1: Nice catch :)